python/amici/ode_export.py

```diff
@@ -1322,9 +1322,11 @@ def _compute_equation(self, name):
         elif name in self._total_derivative_prototypes:
             args = self._total_derivative_prototypes[name]
             args['name'] = name
-            self._lock_total_derivative = True
+            for cv in args['chainvars']:
+                self._lock_total_derivative.add(cv)
```

Comment:
```suggestion
                self._lock_total_derivative |= set(args['chainvars'])
```
Not sure if we need set().